### ISSUE TITLE
Error on compilation

### ISSUE THREAD

**Issue by ZdenekM** [PR2/pr2_simulator#93]
Ubuntu 12.10, ROS Hydro, pr2_simulator branch: hydro-devel

```
In file included from /home/imaterna/ros_hydro/catkin_ws/src/pr2_simulator/pr2_gazebo_plugins/src/gazebo_ros_power_monitor.cpp:38:0:
/home/imaterna/ros_hydro/catkin_ws/src/pr2_simulator/pr2_gazebo_plugins/include/pr2_gazebo_plugins/gazebo_ros_power_monitor.h:37:28: fatal error: physics/World.hh: Adresář nebo soubor neexistuje
compilation terminated.
make[2]: *** [pr2_simulator/pr2_gazebo_plugins/CMakeFiles/gazebo_ros_power_monitor.dir/src/gazebo_ros_power_monitor.cpp.o] Error 1
make[1]: *** [pr2_simulator/pr2_gazebo_plugins/CMakeFiles/gazebo_ros_power_monitor.dir/all] Error 2
```

**Comment by hsu** [comment]
should be addressed by pull request #95

**Comment by ZdenekM** [comment]
Unfortunately, now I'm getting this:

```
-- +++ processing catkin package: 'pr2_gazebo_plugins'
-- ==> add_subdirectory(pr2_simulator/pr2_gazebo_plugins)
CMake Error at /opt/ros/hydro/share/cv_bridge/cmake/cv_bridgeConfig.cmake:141 (message):
  Project 'pr2_gazebo_plugins' tried to find library 'debug'.  The library is
  neither a target nor built/installed properly.  Did you compile project
  'cv_bridge'? Did you find_package() it before the subdirectory containing
  its code is included?
Call Stack (most recent call first):
  /opt/ros/hydro/share/catkin/cmake/catkinConfig.cmake:72 (find_package)
  pr2_simulator/pr2_gazebo_plugins/CMakeLists.txt:7 (find_package)


-- Configuring incomplete, errors occurred!
Invoking "cmake" failed
```

**Comment by ahendrix** [comment]
I'm not seeing this with the latest version of pr2_simulator ( 636ce3f80b336a4051c16d5dd074e816a99cec22 ) and cv_bridge 1.10.15 on Hydro and Ubuntu Precise (12.04).

Is this still a problem on your machine?

**Comment by ZdenekM** [comment]
I'm on 636ce3f and it's still problem. Hydro, Ubuntu Raring (13.04), cv_bridge 1.10.15. I will try on different machine.

**Comment by ahendrix** [comment]
I can't reproduce this on my Raring machine either.

Is it possible that there's another package in your catkin workspace that's causing problems for this?

**Comment by ZdenekM** [comment]
Thanks for hint. I tried to remove build and devel directories and also put CATKIN_IGNORE in all other packages but it still ends with the same error. On different machine it's compiling ok (but dies during launching).

**Comment by ahendrix** [comment]
This definitely sounds like a problem with that particular machine.

Can you open a separate ticket to track the issues you're seeing on launch?

**Comment by beranv** [comment]
Ubuntu 12.10, ROS Hydro, pr2_simulator branch: hydro-devel
`/home/beranv/ros_hydro/catkin_ws/src/pr2_simulator/pr2_gazebo_plugins/src/gazebo_ros_controller_manager.cpp:47:24: fatal error: sdf/Param.hh: No such file or directory
compilation terminated.`

**Comment by ZdenekM** [comment]
@ahendrix Yes, it seems to be a problem with particular machine - on my ntb it works great. But on the PC I'm not able to figure out what is wrong.

**Comment by trainman419** [comment]
@ZdenekM I've seen an issue similar to what you're seeing when I had an upstream package that was exporting libraries poorly. I ended up taking a look at the list of libraries inside of the generated cmake file (in your case /opt/ros/hydro/share/cv_bridge/cmake/cv_bridgeConfig.cmake ) and then spent some time trying to figure out where each of those libraries were coming from and why.

@beranv that sounds like a completely different issue; can you open a new ticket for it?
